<!DOCTYPE html>
<html lang="en"><head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1"><!-- Begin Jekyll SEO tag v2.8.0 -->
<title>Publications | Fernando Zhapa-Camacho</title>
<meta name="generator" content="Jekyll v3.9.2" />
<meta property="og:title" content="Publications" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="…" />
<meta property="og:description" content="…" />
<link rel="canonical" href="http://localhost:4000/publications/" />
<meta property="og:url" content="http://localhost:4000/publications/" />
<meta property="og:site_name" content="Fernando Zhapa-Camacho" />
<meta property="og:type" content="website" />
<meta name="twitter:card" content="summary" />
<meta property="twitter:title" content="Publications" />
<script type="application/ld+json">
{"@context":"https://schema.org","@type":"WebPage","description":"…","headline":"Publications","url":"http://localhost:4000/publications/"}</script>
<!-- End Jekyll SEO tag -->
<link rel="stylesheet" href="/assets/main.css">
  <link rel="stylesheet" href="/assets/css/styles.css">
  <link rel="stylesheet" href="/assets/css/academicons-1.9.3/css/academicons.css"><link type="application/atom+xml" rel="alternate" href="http://localhost:4000/feed.xml" title="Fernando Zhapa-Camacho" /></head>
<body><header class="site-header" role="banner">

  <div class="wrapper"><a class="site-title" rel="author" href="/">Fernando Zhapa-Camacho</a><nav class="site-nav">
        <input type="checkbox" id="nav-trigger" class="nav-trigger" />
        <label for="nav-trigger">
          <span class="menu-icon">
            <svg viewBox="0 0 18 15" width="18px" height="15px">
              <path d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.032C17.335,0,18,0.665,18,1.484L18,1.484z M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.032C17.335,6.031,18,6.696,18,7.516L18,7.516z M18,13.516C18,14.335,17.335,15,16.516,15H1.484 C0.665,15,0,14.335,0,13.516l0,0c0-0.82,0.665-1.483,1.484-1.483h15.032C17.335,12.031,18,12.695,18,13.516L18,13.516z"/>
            </svg>
          </span>
        </label>

        <div class="trigger"><a class="page-link" href="/publications/">Publications</a><a class="page-link" href="/talks/">Talks</a></div>
      </nav></div>
</header>
<main class="page-content" aria-label="Content">
      <div class="wrapper">
        <article class="post">

  <header class="post-header">
    <h1 class="post-title">Publications</h1>
  </header>

  <div class="post-content">
    <br>



<div class="container">
  <h4>CatE: Embedding ALC ontologies using category-theoretical semantics</h4>
  
  <p class="button one">Preprint</p>
  
</div>
<p><i><p><strong>Fernando Zhapa-Camacho</strong>, Robert Hoehndorf</p>
</i></p>

<div class="justified">
  <p> <p>Machine learning with Semantic Web ontologies follows several strategies, 
one of which involves projecting ontologies into graph
structures and applying graph embeddings or graph-based machine
learning methods to the resulting graphs. Several methods have been
developed that project ontology axioms into graphs. However, these
methods are limited in the type of axioms they can project (totality),
whether they are invertible (injectivity), and how they exploit
semantic information. These limitations restrict the kind of tasks to
which they can be applied. Category-theoretical semantics of logic
languages formalizes interpretations using categories instead of sets,
and categories have a graph-like structure. We developed CatE, which
uses the category-theoretical formulation of the semantics of the
Description Logic $\mathcal{ALC}$ to generate a graph representation for ontology
axioms. The CatE projection is total and injective, and therefore
overcomes limitations of other graph-based ontology embedding methods
which are generally not invertible. We apply CatE to a number of
different tasks, including deductive and inductive reasoning, and we
demonstrate that CatE improves over state of the art ontology
embedding methods. Furthermore, we show that CatE can also outperform
model-theoretic ontology embedding methods in machine learning tasks
in the biomedical domain.</p>

 </p>
</div>

<div class="container">
  <a href="" target="_blank" class="button two">
    
    ArXiv
    
  </a>
  
  <a href="https://github.com/bio-ontology-research-group/catE" target="_blank" class="button two">GitHub</a>
  
  


</div>

  <!--    -->
  <!-- <button type="button" class="collapsible button four">Bibtex Citation</button> -->
  <!-- <div class="content"> -->
  <!--   <pre> -->
  <!--     
@misc{zhapacamacho2023cate,
title={CatE: Embedding $\mathcal{ALC}$ ontologies using category-theoretical semantics}, 
author={Fernando Zhapa-Camacho and Robert Hoehndorf},
year={2023},
eprint={2305.07163},
archivePrefix={arXiv},
primaryClass={cs.LO}
}
 -->
  <!--   </pre> -->
  <!-- </div> -->
  <!--  -->
  
  <br>
  <br>


<div class="container">
  <h4>From axioms over graphs to vectors, and back again: evaluating the properties of graph-based ontology embeddings</h4>
  
  
  <a href="https://sites.google.com/view/nesy2023" target="_blank" class="button one">NeSy 2023</a>
  
  
</div>
<p><i><p><strong>Fernando Zhapa-Camacho</strong>, Robert Hoehndorf</p>
</i></p>

<div class="justified">
  <p> <p>Several approaches have been developed that generate embeddings for
Description Logic ontologies and use these embeddings in machine
learning. One approach of generating ontologies embeddings is by first
embedding the ontologies into a graph structure, i.e., introducing a
set of nodes and edges for named entities and logical axioms, and then
applying a graph embedding to embed the graph in ℝn. Methods that
embed ontologies in graphs (graph projections) have different formal
properties related to the type of axioms they can utilize, whether the
projections are invertible or not, and whether they can be applied to
asserted axioms or their deductive closure. We analyze, qualitatively
and quantitatively, several graph projection methods that have been
used to embed ontologies, and we demonstrate the effect of the
properties of graph projections on the performance of predicting
axioms from ontology embeddings. We find that there are substantial
differences between different projection methods, and both the
projection of axioms into nodes and edges as well ontological choices
in representing knowledge will impact the success of using ontology
embeddings to predict axioms.</p>
 </p>
</div>

<div class="container">
  <a href="" target="_blank" class="button two">
    
    ArXiv
    
  </a>
  
  <a href="https://github.com/bio-ontology-research-group/ontology-graph-projections" target="_blank" class="button two">GitHub</a>
  
  


</div>

  <!--    -->
  <!-- <button type="button" class="collapsible button four">Bibtex Citation</button> -->
  <!-- <div class="content"> -->
  <!--   <pre> -->
  <!--     @misc{zhapacamacho2023axioms,
      title={From axioms over graphs to vectors, and back again: evaluating the properties of graph-based ontology embeddings}, 
      author={Fernando Zhapa-Camacho and Robert Hoehndorf},
      year={2023},
      eprint={2303.16519},
      archivePrefix={arXiv},
      primaryClass={cs.AI}
} -->
  <!--   </pre> -->
  <!-- </div> -->
  <!--  -->
  
  <br>
  <br>


<div class="container">
  <h4>mOWL: Python library for machine learning with biomedical ontologies</h4>
  
  
   <a href="https://doi.org/10.1093/bioinformatics/btac811" target="_blank" class="button one">Bioinformatics, Volume 39, Issue 1</a>
  <!-- <p class="button one">Bioinformatics, Volume 39, Issue 1</p> -->
  
  
</div>
<p><i><p><strong>Fernando Zhapa-Camacho</strong>, Maxat Kulmanov, Robert Hoehndorf</p>
</i></p>

<div class="justified">
  <p> <p><strong>Motivation</strong> Ontologies contain formal and structured information
about a domain and are widely used in bioinformatics for annotation
and integration of data. Several methods use ontologies to provide
background knowledge in machine learning tasks, which is of particular
importance in bioinformatics. These methods rely on a set of common
primitives that are not readily available in a software library; a
library providing these primitives would facilitate the use of current
machine learning methods with ontologies and the development of novel
methods for other ontology-based biomedical applications.</p>

<p><strong>Results</strong> We developed mOWL, a Python library for machine learning
with ontologies formalized in the Web Ontology Language (OWL). mOWL
implements ontology embedding methods that map information contained
in formal knowledge bases and ontologies into vector spaces while
preserving some of the properties and relations in ontologies, as well
as methods to use these embeddings for similarity computation,
deductive inference and zero-shot learning. We demonstrate mOWL on the
knowledge-based prediction of protein–protein interactions using the
gene ontology and gene–disease associations using phenotype
ontologies.</p>

 </p>
</div>

<div class="container">
  <a href="https://doi.org/10.1093/bioinformatics/btac811" target="_blank" class="button two">
    
    Paper
    
  </a>
  
  <a href="https://github.com/bio-ontology-research-group/mowl" target="_blank" class="button two">GitHub</a>
  
  


</div>

  <!--    -->
  <!-- <button type="button" class="collapsible button four">Bibtex Citation</button> -->
  <!-- <div class="content"> -->
  <!--   <pre> -->
  <!--     @article{10.1093/bioinformatics/btac811,
	author = {Zhapa-Camacho, Fernando and Kulmanov, Maxat and Hoehndorf, Robert},
    	title = "{mOWL: Python library for machine learning with biomedical ontologies}",
    	journal = {Bioinformatics},
    	year = {2022},
    	month = {12},
    	issn = {1367-4803},
    	doi = {10.1093/bioinformatics/btac811},
    	url = {https://doi.org/10.1093/bioinformatics/btac811},
    	note = {btac811},
   	} -->
  <!--   </pre> -->
  <!-- </div> -->
  <!--  -->
  
  <br>
  <br>


<div class="container">
  <h4>A-LIOn - Alignment Learning through Inconsistency negatives of the aligned Ontologies</h4>
  
  
  <a href="http://om2022.ontologymatching.org/" target="_blank" class="button one">The 17th International Workshop on Ontology Matching</a>
  
  
</div>
<p><i><p>Sarah M. Alghamdi, <strong>Fernando Zhapa-Camacho</strong>, Robert Hoehndorf</p>
</i></p>

<div class="justified">
  <p> <p>Ontologies play an important role in sharing and reusing knowledge. Several ontologies have been
developed to describe a particular domain but from different perspectives from communities of developers
and users. This has led to the existence of multiple ontologies covering the same or a different domain
with varying degrees of variability. Ontology Alignment is typically used to identify correspondences
between semantically related elements of two or more ontologies in order to address this problem.
We propose A-LIOn a system that learns alignments by combining lexical and semantic approaches
as well as machine learning. The system utilizes OWL EL reasoning for negative sampling which is
iteratively used to inform the correction of the learning of the alignments. We demonstrate that A-LIOn
produces alignments that are coherent with respect to OWL EL.</p>
 </p>
</div>

<div class="container">
  <a href="https://ceur-ws.org/Vol-3324/oaei22_paper2.pdf" target="_blank" class="button two">
    
    Paper
    
  </a>
  
  <a href="https://github.com/bio-ontology-research-group/a-lion" target="_blank" class="button two">GitHub</a>
  
  


</div>

  <!--    -->
  
  <br>
  <br>


<div class="container">
  <h4>DeepGOWeb: fast and accurate protein function prediction on the (Semantic) Web</h4>
  
  
   <a href="https://doi.org/10.1093/nar/gkab373" target="_blank" class="button one">Nucleic Acids Research</a>
  <!-- <p class="button one">Nucleic Acids Research</p> -->
  
  
</div>
<p><i><p>Maxat Kulmanov, <strong>Fernando Zhapa-Camacho</strong>, Robert Hoehndorf</p>
</i></p>

<div class="justified">
  <p> <p>Understanding the functions of proteins is crucial to understand
biological processes on a molecular level. Many more protein sequences
are available than can be investigated experimentally. DeepGOPlus is a
protein function prediction method based on deep learning and sequence
similarity. DeepGOWeb makes the prediction model available through a
website, an API, and through the SPARQL query language for
interoperability with databases that rely on Semantic Web
technologies. DeepGOWeb provides accurate and fast predictions and
ensures that predicted functions are consistent with the Gene
Ontology; it can provide predictions for any protein and any function
in Gene Ontology. DeepGOWeb is freely available at
(https://deepgo.cbrc.kaust.edu.sa/)</p>
 </p>
</div>

<div class="container">
  <a href="https://doi.org/10.1093/nar/gkab373" target="_blank" class="button two">
    
    Paper
    
  </a>
  
  


</div>

  <!--    -->
  <!-- <button type="button" class="collapsible button four">Bibtex Citation</button> -->
  <!-- <div class="content"> -->
  <!--   <pre> -->
  <!--     @article{10.1093/nar/gkab373,
    author = {Kulmanov, Maxat and Zhapa-Camacho, Fernando and Hoehndorf, Robert},
    title = "{DeepGOWeb: fast and accurate protein function prediction on the (Semantic) Web}",
    journal = {Nucleic Acids Research},
    volume = {49},
    number = {W1},
    pages = {W140-W146},
    year = {2021},
    month = {05},
    issn = {0305-1048},
    doi = {10.1093/nar/gkab373},
    url = {https://doi.org/10.1093/nar/gkab373},
    eprint = {https://academic.oup.com/nar/article-pdf/49/W1/W140/38841723/gkab373.pdf},
} -->
  <!--   </pre> -->
  <!-- </div> -->
  <!--  -->
  
  <br>
  <br>


<div class="container">
  <h4>Successive Adaptive Linear Neural Modeling for Equidistant Real Roots Finding</h4>
  
  
  <a href="https://attend.ieee.org/etcm-2018/" target="_blank" class="button one">ETCM 2018</a>
  
  
</div>
<p><i><p>Joseph R. González, <strong>Fernando Zhapa-Camacho</strong>, Oscar V. Guarnizo, Francisco Ortega-Zamorano</p>
</i></p>

<div class="justified">
  <p> <p>The main objective of this work has been to implement a model to find
equidistant real roots using a Successive Adaptive Linear Neural
Modeling which uses two approaches: a Self Organized Map (SOM) and an
Adaptative Linear Neuron (Adaline). A SOM model has been used with a
new neighborhood function Λ, and a physical distance β with which the
task is divided in sub-processes reducing the complexity of the task
because the SOM model can delimited the areas where a single root
exist. Then, through a successive approach, it is applied an
Feed-forward neural model with a learning process base on Adaline
neuron with pocket in each pair of regions for finding the real root
values with a reduced precision. Finally, several experiments were
done consider CPU time, relative error, distance between the roots and
polynomial degrees. The results show that the time complexity grows in
a linear or logarithmic way. Also, the error does not increase in a
higher rate than the degree of polynomial or the root distance.</p>
 </p>
</div>

<div class="container">
  <a href="https://doi.org/10.1109/ETCM.2018.8580280" target="_blank" class="button two">
    
    Paper
    
  </a>
  
  


</div>

  <!--    -->
  <!-- <button type="button" class="collapsible button four">Bibtex Citation</button> -->
  <!-- <div class="content"> -->
  <!--   <pre> -->
  <!--     @INPROCEEDINGS{8580280,
  author={González, Joseph R. and Zhapa, Fernando P. and Guarnizo, Oscar V. and Ortega-Zamorano, Francisco},
  booktitle={2018 IEEE Third Ecuador Technical Chapters Meeting (ETCM)}, 
  title={Successive Adaptive Linear Neural Modeling for Equidistant Real Roots Finding}, 
  year={2018},
  volume={},
  number={},
  pages={1-6},
  doi={10.1109/ETCM.2018.8580280}} -->
  <!--   </pre> -->
  <!-- </div> -->
  <!--  -->
  
  <br>
  <br>


<script>
var coll = document.getElementsByClassName("collapsible");
var i;

for (i = 0; i < coll.length; i++) {
  coll[i].addEventListener("click", function() {
    this.classList.toggle("active");
    var content = this.nextElementSibling;
    if (content.style.maxHeight){
      content.style.maxHeight = null;
    } else {
      content.style.maxHeight = content.scrollHeight + "px";
    } 
  });
}
</script>


  </div>

</article>

      </div>
    </main><footer class="site-footer h-card">
  <data class="u-url" href="/"></data>

  <div class="wrapper">

    <h2 class="footer-heading">Fernando Zhapa-Camacho</h2>

    <div class="footer-col-wrapper">
      <div class="footer-col footer-col-1">
        <ul class="contact-list">
          <li class="p-name">Fernando Zhapa-Camacho</li><li><a class="u-email" href="mailto:ferzcam@gmail.com, fernando.zhapacamacho@kaust.edu.sa">ferzcam@gmail.com, fernando.zhapacamacho@kaust.edu.sa</a></li></ul>
      </div>

      <div class="footer-col footer-col-2"><ul class="social-media-list"><li>
    <a href="https://www.twitter.com/ferzcam">
      <svg class="svg-icon">
	<use xlink:href="/assets/minima-social-icons.svg#twitter">
	</use>
      </svg>
    </a>
  </li><li><a href="https://github.com/ferzcam">
      <svg class="svg-icon">
	<use xlink:href="/assets/minima-social-icons.svg#github">
	</use>
      </svg>
    </a>
  </li><li><a href="https://scholar.google.com/citations?user=tGB1xoAAAAAJ">
      <i class="ai ai-google-scholar-square ai-lg"></i>
    </a>
  </li><li><a href="https://orcid.org/0000-0002-0710-2259">
      <i class="ai ai-orcid-square ai-lg"></i>
    </a>
  </li><li><a href="https://www.researchgate.net/profile/Fernando-Zhapa-2">
      <i class="ai ai-researchgate-square ai-lg"></i>
    </a>
  </li></ul>
</div>

      <div class="footer-col footer-col-3">
        <p>...</p>
      </div>
    </div>

  </div>

</footer>
</body>

</html>
